bundles/org.openhab.core.automation/src/main/java/org/openhab/core/automation/util/ActionInputsHelper.java

```diff
@@ -0,0 +1,327 @@
+                    // A text parameter is used. The expected value must respect a particular format specific
+                    // to each of these types.
+                    break;
+                default:
+                    supported = false;
+                    break;
+            }
+        }
+        if (!supported) {
+            throw new IllegalArgumentException("Input parameter '" + input.getName() + "' with type " + input.getType()
+                    + "cannot be converted into a config description parameter!");
+        }
+
+        ConfigDescriptionParameterBuilder builder = ConfigDescriptionParameterBuilder
+                .create(input.getName(), parameterType).withLabel(input.getLabel())
+                .withDescription(input.getDescription()).withReadOnly(false)
+                .withRequired(required || input.isRequired()).withContext(context);
+        if (!input.getDefaultValue().isEmpty()) {
+            builder = builder.withDefault(input.getDefaultValue());
+        } else if (defaultValue != null) {
+            builder = builder.withDefault(defaultValue);
+        }
+        if (unit != null) {
+            builder = builder.withUnit(unit.getSymbol());
+        }
+        return builder.build();
+    }
+
+    /**
+     * Maps serialised inputs to the Java types required by the {@link Input}s of the given {@link ActionType}.
+     *
+     * @param actionType the action type whose inputs to consider
+     * @param arguments the serialised arguments
+     * @return the mapped arguments
+     */
+    public Map<String, Object> mapSerializedInputsToActionInputs(ActionType actionType, Map<String, Object> arguments) {
+        Map<String, Object> newArguments = new HashMap<>();
+        for (Input input : actionType.getInputs()) {
+            String name = input.getName();
+            Object value = arguments.get(name);
+            try {
+                value = mapSerializedInputToActionInput(actionType, input, value);
+            } catch (IllegalArgumentException e) {
+                logger.warn("{} Input parameter is ignored.", e.getMessage());
+                value = null;
+            }
+            if (value == null) {
+                continue;
+            }
+            newArguments.put(name, value);
+        }
+        return newArguments;
+    }
+
+    /**
+     * Maps a serialised input to the Java type required by the given {@link Input}.
+     *
+     * @param actionType the action type whose inputs to consider
+     * @param input the input whose type to consider
+     * @param argument the serialised argument
+     * @return the mapped argument or null if the input argument was null
+     * @throws IllegalArgumentException if the mapping failed
+     */
+    public @Nullable Object mapSerializedInputToActionInput(ActionType actionType, Input input,
+            @Nullable Object argument) throws IllegalArgumentException {
+        boolean failed = false;
```

Comment:
If I'm not mistaken `failed` is set to `true` only in case of an `IllegalArgumentException` or a `NumberFormatException`. You can put a `try-catch` around the whole method and throw the new `IllegalArgumentException` in the `catch`-block. `failed` is not needed then and you simplify the code a lot.

Reply:
Done